packages/server/src/integrations/googlecloud.ts

```diff
@@ -0,0 +1,173 @@
+      required: true,
+    },
+    privateKey: {
+      type: DatasourceFieldType.LONGFORM,
+      required: true,
+    },
+    clientEmail: {
+      type: DatasourceFieldType.STRING,
+      required: true,
+    },
+  },
+  query: {
+    create: {
+      type: QueryType.FIELDS,
+      fields: {
+        bucket: {
+          display: "New Bucket",
+          type: DatasourceFieldType.STRING,
+          required: true,
+        },
+        location: {
+          required: true,
+          type: DatasourceFieldType.SELECT,
+          display: "Location",
+          default: "US",
+          config: {
+            options: ["ASIA", "EU", "US"],
+          },
+        },
+        storageClass: {
+          required: true,
+          type: DatasourceFieldType.SELECT,
+          display: "Storage Class",
+          default: "STANDARD",
+          config: {
+            options: ["STANDARD", "NEARLINE", "COLDLINE", "ARCHIVE"],
+          },
+        },
+        allowPublicAccess: {
+          type: DatasourceFieldType.BOOLEAN,
+          display: "Allow Public Access",
+          default: false,
+        },
+      },
+    },
+    read: {
+      type: QueryType.FIELDS,
+      fields: {
+        bucket: {
+          type: DatasourceFieldType.STRING,
+          required: true,
+        },
+      },
+    },
+    delete: {
+      type: QueryType.FIELDS,
+      fields: {
+        bucket: {
+          type: DatasourceFieldType.STRING,
+          required: true,
+        },
+        delete: {
+          type: DatasourceFieldType.STRING,
+          required: true,
+        },
+      },
+    },
+  },
+}
+
+class GoogleCloudIntegration implements IntegrationBase {
+  private readonly config: GoogleCloudConfig
+  private storage: any
+
+  constructor(config: GoogleCloudConfig) {
+    this.config = config
+    this.storage = new Storage({
+      projectId: this.config.projectId,
+      scopes: "https://www.googleapis.com/auth/cloud-platform",
+      credentials: {
+        client_email: this.config.clientEmail,
+        private_key: this.config.privateKey.split(String.raw`\n`).join("\n"),
+      },
+    })
+  }
+
+  async testConnection() {
+    const response: ConnectionInfo = {
+      connected: false,
+    }
+    try {
+      await this.connect()
+      return { connected: true }
+    } catch (e: any) {
+      return {
+        connected: false,
+        error: e.message as string,
+      }
+    }
+  }
+
+  async connect() {
+    await this.storage.getBuckets()
+  }
+
+  async read(query: { bucket: string }) {
+    const bucket = this.storage.bucket(query.bucket)
+    // Does support paging. Could possibly use this feature
+    const [files, queryForPage2] = await bucket.getFiles({
+      autoPaginate: false,
+    })
+    return files.map((file: any) => file.metadata)
+  }
+
+  async delete(query: { bucket: string; delete: string }) {
+    return await this.storage.bucket(query.bucket).file(query.delete).delete()
+  }
+
+  async create(query: {
```

Comment:
Generally you want to annotate methods that override something in a base class with the `override` keyword, e.g.

```typescript
  async override create(query: {
```

Doing this allows the TypeScript compiler to check for you that the method is overridden correctly. 🙂

Reply:
Does this apply to classes that implement an interface? I added the keyword but VS complains that it can't be used as `'GoogleCloudIntegration' does not extend another class`

Reply:
Ah, sorry, it does not. My mistake!